lobsterpy/structuregraph/__init__.py

```diff
@@ -0,0 +1,154 @@
+            chemenvlobster = LobsterNeighbors(
+                are_coops=False,
+                filename_ICOHP=self.path_to_icohplist,
+                structure=Structure.from_file(self.path_to_poscar),
+                additional_condition=0,
+                filename_CHARGE=self.path_to_charge,
+                add_additional_data_sg=self.add_additional_data_sg,
+                filename_blist_sg1=self.path_to_icobilist,
+                id_blist_sg1="ICOBI",
+                filename_blist_sg2=self.path_to_icooplist,
+                id_blist_sg2="ICOOP",
+            )
+
+        else:
+
+            chemenvlobster = LobsterNeighbors(
+                are_coops=False,
+                filename_ICOHP=self.path_to_icohplist,
+                structure=Structure.from_file(self.path_to_poscar),
+                additional_condition=0,
+                filename_CHARGE=self.path_to_charge,
+                add_additional_data_sg=self.add_additional_data_sg,
+            )
+
+        chemenvlobster.get_info_cohps_to_neighbors(path_to_COHPCAR=self.path_to_cohpcar)
+
+        (
+            _summed_icohps,
+            list_icohps,
+            _number_bonds,
+            labels,
+            _atoms,
+            _final_isites,
+        ) = chemenvlobster.get_info_icohps_to_neighbors()
+
+        decorated_structure = Charge(self.path_to_charge).get_structure_with_charges(
+            self.path_to_poscar
+        )
+
+        lobster_env = chemenvlobster.get_bonded_structure(
+            structure=decorated_structure, decorate=True, edge_properties=True
+        )
+
+        analyze = Analysis(
+            path_to_charge=self.path_to_charge,
+            path_to_cohpcar=self.path_to_cohpcar,
+            path_to_poscar=self.path_to_poscar,
+            path_to_icohplist=self.path_to_icohplist,
+            path_to_madelung=self.path_to_madelung,
+            whichbonds="all",
+        )
+
+        cba = analyze.condensed_bonding_analysis
+
+        for k, v in cba["sites"].items():
+            for k2, v2 in lobster_env.graph.nodes.data():
+                if v["ion"] == v2["specie"]:
+                    v2["properties"].update({"env": v["env"]})
+
+        for edge_prop in lobster_env.graph.edges.data():
+            for i, icohp in enumerate(list_icohps):
+                if icohp == edge_prop[2]["ICOHP"]:
+                    _ab, ab_p, _b, b_p = analyze._integrate_antbdstates_below_efermi(
+                        cohp=chemenvlobster.completecohp.get_cohp_by_label(labels[i]),
+                        start=None,
+                    )
+                    edge_prop[2]["ICOHP_bonding_perc"] = b_p
+                    edge_prop[2]["ICOHP_antibonding_perc"] = ab_p
+
+        return lobster_env
+
+    def get_sg_node_data(self):
+        """
+        Method of extract node data of graph object
+        Returns:
+           will return a list consisting of node data
+        """
+
+        return print(self.sg.graph.nodes(data=True))
```

Comment:
I am not sure we need this. I would only like to have a method that just returns the new structure graph. The printing can be written easily by a user.